client/consul.go

```diff
@@ -256,78 +268,72 @@ func (c *ConsulService) registerService(service *structs.Service, task *structs.
 		Address: host,
 	}
 
-	if err := c.client.Agent().ServiceRegister(asr); err != nil {
+	if err := c.client.ServiceRegister(asr); err != nil {
 		c.logger.Printf("[DEBUG] consul: Error while registering service %v with Consul: %v", service.Name, err)
 		mErr.Errors = append(mErr.Errors, err)
 	}
-	checks := c.makeChecks(service, host, port)
-	for _, check := range checks {
-		if err := c.registerCheck(check); err != nil {
+	for _, check := range service.Checks {
+		cr := c.makeCheck(service, check, host, port)
+		if err := c.registerCheck(cr); err != nil {
```

Comment:
@dadgar yeah that's fine, we are just swallowing the error. PerformSync keeps trying to register with Consul every 5 seconds.